vendor/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+
+if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+# Update submodules as needed
+    option(GIT_SUBMODULE "Check submodules during build" ON)
+    if(GIT_SUBMODULE)
+        message(STATUS "Submodule update")
+        execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
+                        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                        RESULT_VARIABLE GIT_SUBMOD_RESULT)
+        if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+            message(FATAL_ERROR "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+        endif()
+    endif()
+endif()
+
+
+add_subdirectory(btwxt)
```

Comment:
You need to guard against the possibility that btwxt is already a dependency in the parent program. See this example for [fmt in Kiva](https://github.com/bigladder/kiva/blob/eff0313b15cd48920a6bba404241913493269658/vendor/CMakeLists.txt#L4-L6).

Reply:
Got it. @chipbarnaby  This should prevent HPWHsim from duplicating btwxt builds in CSE.